### ISSUE TITLE
Add ClientBuilder

### ISSUE THREAD

**Issue by Xuanwo** [Xuanwo/hdrs#111]
There many things for client to configure. It's better to introduce a ClientBuilder instead.

Also fix https://github.com/Xuanwo/hdrs/issues/19

**Comment by zuston** [comment]
I want to pick up this and support kerberos @Xuanwo

**Comment by Xuanwo** [comment]
Great!

I expect to have the following API:

```rust
struct ClientBuilder { ... }
impl ClientBuilder {
   fn new(name_node: &str) -> ClientBuilder
   fn with_user(mut self, user: &str) -> ClientBuilder
   fn with_xxxx(mut self, v: yyy) -> ClientBuilder
   fn connect(self) -> Result<Client>
}
```

After `ClientBuilder` introduced, we will remove `Client::connect()` and `Client::connect_as_user()`.

**Comment by Xuanwo** [comment]
Implemented by https://github.com/Xuanwo/hdrs/pull/112, thanks @zuston

**Comment by zuston** [comment]
> Implemented by #112, thanks @zuston

Thanks @Xuanwo for picking up. I will test it ASSP
